code/controllers/subsystem/persistence/circuits.dm

```diff
@@ -0,0 +1,32 @@
+#define CIRCUITS_DATA_FILEPATH "data/circuit_designs.json"
+
+/datum/controller/subsystem/persistence/proc/load_circuits()
+	var/json_file = file(CIRCUITS_DATA_FILEPATH)
+	if(!fexists(json_file))
+		return
+	var/list/json = json_decode(file2text(json_file))
+	if(!json)
+		return
+	circuit_designs = json["data"]
+	for (var/list/design in circuit_designs)
+		var/list/new_materials = list()
+		for (var/material in design["materials"])
+			new_materials[GET_MATERIAL_REF(text2path(material))] = design["materials"][material]
+		design["materials"] = new_materials
+
+/datum/controller/subsystem/persistence/proc/save_circuits()
```

Comment:
IMO this should either be per ckey, so that we can load just that ckey into memory, or json_database.